### PR TITLE
Feature/core 10473 folder events geo tree

### PR DESCRIPTION
**Pull request by Navas23**

### CODE REVIEW

File: src/elements/GeoTree/GeoTreeItem.vue [hunk 1]
```diff
@@ -244,19 +245,25 @@ export default {
...
+      this.$emit('check-folder', category, isChecked, isDelegated)
     },
     /**
      * To check all items of a category
      */
-    handleCheck (category, isChecked) {
+    handleCheck (category, isChecked, isDelegated = false) {
       if (category[this.keyForSubcategory]) {
+        if (_.size(category[this.keyForSubcategory])) { // We don't want to emit event for empty folders
+          this.$emit('check-folder', _.omit(category, this.keyForSubcategory), isChecked, isDelegated)
```
Comment (fofi) [inline src/elements/GeoTree/GeoTreeItem.vue:260]: Why do we want to omit the keyForSubcategory ?

**Review by fofi — APPROVED** [review]

File: src/elements/GeoTree/GeoTreeItem.vue (hunk 1 above)
Comment (Navas23) [reply src/elements/GeoTree/GeoTreeItem.vue:260]: We could we sending a huge object there that we are not going to use, IMO it doesn't make sense to send all the subfolder and subitems

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 33.0.0
+New:
```
Comment (dletran) [inline CHANGELOG.md:4]: small thing: this is a breaking change not just new right?

**Review by dletran — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (Navas23) [reply CHANGELOG.md:4]: It's breaking and new, I will update it

File: src/elements/GeoTree/GeoTreeItem.vue (hunk 1 above)
Comment (irmakyz) [reply src/elements/GeoTree/GeoTreeItem.vue:260]: makes sense for me

**Pull request merged**